exchanges/btse/btse_wrapper.go

```diff
@@ -397,7 +393,41 @@ func (b *BTSE) GetFundingHistory() ([]exchange.FundHistory, error) {
 
 // GetExchangeHistory returns historic trade data within the timeframe provided.
 func (b *BTSE) GetExchangeHistory(p currency.Pair, assetType asset.Item, timestampStart, timestampEnd time.Time) ([]exchange.TradeHistory, error) {
-	return nil, common.ErrNotYetImplemented
+	if assetType != asset.Spot {
+		return nil, common.ErrNotYetImplemented
+	}
+
+	fpair, err := b.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+
+	trades, err := b.GetTrades(fpair.String(),
+		timestampStart, timestampEnd,
+		0, 0, 0,
+		false)
+	if err != nil {
+		return nil, err
+	}
+	var resp []exchange.TradeHistory
+	for x := range trades {
+		tempExch := exchange.TradeHistory{
+			Timestamp: trades[x].Time,
+			Price:     trades[x].Price,
+			Amount:    trades[x].Amount,
+			Exchange:  b.Name,
+			Type:      trades[x].Type,
+			TID:       strconv.Itoa(trades[x].SerialID),
+		}
+		resp = append(resp, tempExch)
+	}
+	return resp, nil
+}
+
+func (b *BTSE) withinLimits(pair string, amount float64) bool {
+	return (math.Mod(amount, b.OrderSizeLimits[pair].MinSizeIncrement) == 0) ||
```

Comment:
I think this should be protected by a rwmutex similar to the Kraken asset translator. Difference being, Kraken wrappers perform numerous lookups so there's a high chance of a concurrent read event whereas here there's only one func doing lookups. What do you think?

Reply:
I was considering converting it to sync.Map since it's basicly what sync.Map was designed for (one write many reads) and will stop mutex contention

Reply:
Sounds good!